### DISCUSSION TITLE
Why the sampler only use the real part of the proposed pdf?

### DISCUSSION THREAD

**Discussion by yiminghwang** [netket/netket#1314] [Q&A] [answered]
Hi, there,

I'm reading the sampler part the document, there is one thing I am curious about is why the netket only takes the real part of the proposed pdf to sample.

Suppose we use the NN to construct the amplitude, such as $f(\sigma)= re(\sigma)+1j*im(\sigma)$, where $re()$ and $im()$ are NN.
The sampler will generate a set of samples $\sigma$ that respects the following condition:
$p(\sigma)=|f(\sigma)|^\alpha \sim \frac{\exp[\alpha\log(f(\sigma))]}{\sum_{\zeta\in H}\exp[\alpha\log(f(\zeta)]}$

However, when I read the source code and the document about the sampler. I found you are only using the real part to calculate,
https://netket.readthedocs.io/en/latest/docs/sampler.html, constructing samplers section,
$p(\sigma)=|f(\sigma)|^\alpha \sim \frac{\exp[\alpha\mathbb{R}[\log(f(\sigma))]]}{\sum_{\zeta\in H}\exp[\alpha\mathbb{R}[\log(f(\zeta)]]}$
in the source code,
```python
proposal_log_prob = sampler.machine_pow * machine.apply(parameters, σp).real
```

Thus, I am curious about how to handle that our proposed amplitude contains the imaginary part.

**Comment by gcarleo** [comment, marked as answer]
`machine.apply` returns the logarithm of the wave function, that's why we take the real part, it would not be possible to sample from a complex probability distribution...

**Reply by PhilipVinc** [reply]
equivalently, we are using the fact that $|x| =\exp[\Re[\log[x]]]$.
The two equations for $p(\sigma)$ you wrote are equivalent.

**Reply by yiminghwang** [reply]
Thank you both for your quick response!
